mlir/tools/mlir-tblgen/DialectGen.cpp

```diff
@@ -181,6 +196,37 @@ static const char *const operationInterfaceFallbackDecl = R"(
                                       mlir::OperationName opName) override;
 )";
 
+/// The code block for the discardable attribute helper.
+static const char *const discardableAttrHelperDecl = R"(
+    /// Helper to manage the discardable attribute `{1}`.
+    class {0}AttrHelper {{
+      mlir::StringAttr name;
+    public:
+      static constexpr llvm::StringLiteral getNameStr() {{
```

Comment:
What's the benefit of this "helper" pattern over just providing the string name directly?

Reply:
Aren't we already avoiding having raw strings spreaded through the codebase and use symbols instead?

Here we could generate separately a `constexpr StringLiteral kMyDiscardableAttrName = "the dialect.my_discardable_attr";` ; but since we're already having a class for the rest of the functionality it seemed better to group it all together.

How would you do this?

Reply:
I guess my only complaint is that the helper pattern is a bit verbose, but it's efficient because it enables one to avoid deferencing the dialect pointer more than once. Seems fine to me.